src/supervisor3.erl

```diff
@@ -172,11 +172,11 @@
 -endif.
 
 -record(state, {name,
-                strategy               :: strategy(),
```

Comment:
Please move all field specs from the record definition to the definition of `-type state() :: ...`. This code change makes it legal that "half of record can be initialized", but as far as I follow the code, it shouldn't be legal.